fgvc/core/models.py

```diff
@@ -34,42 +36,77 @@ def get_model(
     Returns
     -------
     model
-        PyTorch `nn.Module` instance.
+        PyTorch model from `timm` library.
     """
     model = timm.create_model(architecture_name, pretrained=pretrained and checkpoint_path is None)
 
+    # load custom weights
+    if checkpoint_path is not None:
+        weights = torch.load(checkpoint_path, map_location="cpu")
+
+        # remove prefix "module." created by nn.DataParallel wrapper
+        if all([k.startswith("module.") for k in weights.keys()]):
+            weights = OrderedDict({k[7:]: v for k, v in weights.items()})
+
+        # identify target size in the weights
+        cls_name = model.default_cfg["classifier"]
```

Comment:
I added more test cases, and solved issues with models like DINO-based ViT.

Our method should be now more robust than the `timm`'s `create_model` because it checks weight sizes of classification head in the checkpoint file and automatically adjusts the model to load `state_dict`.